core/src/main/java/hudson/model/AbstractBuild.java

```diff
@@ -176,6 +176,7 @@ protected AbstractBuild(P project, File buildDir) throws IOException {
         super(project, buildDir);
     }
 
+    @Exported
```

Comment:
This limits it to things that inherit from AbstractBuild. Meaning all pipeline jobs will not appear.
Maybe put it on the getParent of the Run class